### PR TITLE
fix(select): Fix Typography component is not centered in the Select component

### LINKED ISSUE #40421: Text alignment issue with Typography component in Select component
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/basic-usage-antd-5-1-6-forked-izq79l?file=/demo.tsx)

image

### Steps to reproduce

- Set value with antd typography component for Select component
- Select the value
- Text is not aligned center

### What is expected?

Text alignment is centered

### What is actually happening?

Text alignment is not centered

| Environment | Info |
| --- | --- |
| antd | 5.1.6 |
| React | 18.2.0 |
| System | macOS Ventura |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by Yuiai01**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/40421

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Fix Typography component is not centered in the Select component         |
| 🇨🇳 Chinese |  修复 `Typography`  在 `Select`  中不居中     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by zombieJ** [comment]
This change breaks origin display logic. Please check:

截屏2023-01-29 22 57 24

File: components/select/style/index.tsx [hunk 1]
```diff
@@ -144,6 +144,8 @@ const genBaseStyle: GenerateStyle<SelectToken> = (token) => {
 
       // ======================== Selection ========================
       [`${componentCls}-selection-item`]: {
+        display: 'flex',
+        alignItems: 'center',
```
Comment (MadCcc) [inline components/select/style/index.tsx:148]:
```suggestion
        alignItems: 'baseline',
```
Try `baseline` to align text.

**Comment by Yuiai01** [comment]
> This change breaks origin display logic. Please check:
> 
> 截屏2023-01-29 22 57 24
> 
> ref: https://app.argos-ci.com/ant-design/ant-design/builds/5619/36429001

嗯嗯 大佬我发现了，由于设置了 flex 布局导致文本省略失效了

File: components/select/style/index.tsx (hunk 1 above)
Comment (Yuiai01) [reply components/select/style/index.tsx:148]: 大佬我刚刚尝试了 `baseline`  ，但是文字省略依然失效

**Comment by Yuiai01** [comment]
> 这里似乎不需要了，看看？[Yuiai01#1](https://github.com/Yuiai01/ant-design/pull/1)

嗯嗯

**Comment by MadCcc** [comment]
放个 debug 用例吧

**Comment by Yuiai01** [comment]
> 放个 debug 用例吧

好了大佬

**Review by MadCcc — APPROVED** [review]

**Comment by LongHaoo** [comment]
这样修复好像有问题 @Yuiai01 ，> * 这样所有的子元素都会被强行 line-height = 30 了，如果我是一写 Tag 的元素，会被直接高度拉满

**Pull request merged**